launcher/src/main/java/org/apache/spark/launcher/AbstractCommandBuilder.java

```diff
@@ -295,7 +295,11 @@ Properties loadPropertiesFile() throws IOException {
       FileInputStream fd = null;
       try {
         fd = new FileInputStream(propsFile);
-        props.load(new InputStreamReader(fd, "UTF-8"));
+        Properties rawProps = new Properties();
```

Comment:
This is way more complicated than it needs to be. Just do:

    for (Map.Entry<Object, Object> e : props.entrySet()) {
      e.setValue(e.getValue().toString().trim());
    }